plane/src/types/backend_state.rs

```diff
@@ -173,9 +178,62 @@ impl Display for BackendStatus {
     }
 }
 
+/// A timestamped representation of a backend's status, along with
+/// termination information. This is used for public-facing endpoints.
+/// It does not include the backend's address, which is only available
+/// to the controller.
 #[derive(Clone, Serialize, Deserialize, Debug)]
-pub struct TimestampedBackendStatus {
+pub struct BackendStatusStreamEntry {
     pub status: BackendStatus,
 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub termination_reason: Option<TerminationReason>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub termination_kind: Option<TerminationKind>,
+
+    /// Whether the process exited with an error. None if the process
+    /// is still running.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub exit_error: Option<bool>,
```

Comment:
Do we want this to include the actual exit code, not just whether an error occurred?

Reply:
I guess the idea is we don't want to expose exit codes for backends on an unauthed endpoint?

If we do decide to leave the exit code out, then, do we add it to the logs like we did in plane1?

Reply:
(Discussed offline)